src/main/java/com/mg105/outputds/BattleCharacterDetails.java

```diff
@@ -0,0 +1,89 @@
+package com.mg105.outputds;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A struct like class that represents a BattleCharacter
+ */
+
+public class BattleCharacterDetails {
+
+    private final String NAME;
```

Comment:
these are not true constants and so they should be `camelCase`

Reply:
(AFAIK `ALL_CAPS_WITH_UNDERSCORES` is reserved for only constants that are final and static)